stdlib/universe/fill.go

```diff
@@ -0,0 +1,251 @@
+		case semantic.Float:
+			spec.Value = strconv.FormatFloat(val.Float(), 'f', -1, 64)
+		case semantic.String:
+			spec.Value = val.Str()
+		case semantic.Time:
+			spec.Value = val.Time().String()
+		default:
+			return nil, errors.New("value type for fill must be a valid primitive type (bool, int, uint, float, string, time)")
+		}
+
+	}
+
+	return spec, nil
+}
+
+func newFillOp() flux.OperationSpec {
+	return new(FillOpSpec)
+}
+
+func (s *FillOpSpec) Kind() flux.OperationKind {
+	return FillKind
+}
+
+type FillProcedureSpec struct {
+	plan.DefaultCost
+	Column      string
+	Value       interface{}
+	UsePrevious bool
+}
+
+func newFillProcedure(qs flux.OperationSpec, pa plan.Administration) (plan.ProcedureSpec, error) {
+	spec, ok := qs.(*FillOpSpec)
+	if !ok {
+		return nil, fmt.Errorf("invalid spec type %T", qs)
+	}
+
+	pspec := &FillProcedureSpec{
+		Column:      spec.Column,
+		UsePrevious: spec.UsePrevious,
+	}
+
+	switch spec.Type {
+	case "bool":
+		v, err := strconv.ParseBool(spec.Value)
+		if err != nil {
+			return nil, err
+		}
+		pspec.Value = v
+	case "int":
+		v, err := strconv.ParseInt(spec.Value, 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		pspec.Value = v
+	case "uint":
+		v, err := strconv.ParseUint(spec.Value, 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		pspec.Value = v
+	case "float":
+		v, err := strconv.ParseFloat(spec.Value, 64)
+		if err != nil {
+			return nil, err
+		}
+		pspec.Value = v
+	case "string":
+		pspec.Value = spec.Value
+	case "time":
+		v, err := values.ParseTime(spec.Value)
+		if err != nil {
+			return nil, err
+		}
+		pspec.Value = v
+	default:
+		return nil, errors.New("unknown type in fill op-spec")
+	}
+
+	return pspec, nil
+}
+
+func (s *FillProcedureSpec) Kind() plan.ProcedureKind {
+	return FillKind
+}
+func (s *FillProcedureSpec) Copy() plan.ProcedureSpec {
+	ns := new(FillProcedureSpec)
+
+	*ns = *s
+
+	return ns
+}
+
+func createFillTransformation(id execute.DatasetID, mode execute.AccumulationMode, spec plan.ProcedureSpec, a execute.Administration) (execute.Transformation, execute.Dataset, error) {
+	s, ok := spec.(*FillProcedureSpec)
+	if !ok {
+		return nil, nil, fmt.Errorf("invalid spec type %T", spec)
+	}
+	cache := execute.NewTableBuilderCache(a.Allocator())
+	d := execute.NewDataset(id, mode, cache)
+	t := NewFillTransformation(d, cache, s)
+	return t, d, nil
+}
+
+type fillTransformation struct {
+	d     execute.Dataset
+	cache execute.TableBuilderCache
+
+	spec *FillProcedureSpec
+}
+
+func NewFillTransformation(d execute.Dataset, cache execute.TableBuilderCache, spec *FillProcedureSpec) *fillTransformation {
+	return &fillTransformation{
+		d:     d,
+		cache: cache,
+		spec:  spec,
+	}
+}
+
+func (t *fillTransformation) RetractTable(id execute.DatasetID, key flux.GroupKey) error {
+	return t.d.RetractTable(key)
+}
+
+func (t *fillTransformation) Process(id execute.DatasetID, tbl flux.Table) error {
+	builder, created := t.cache.TableBuilder(tbl.Key())
```

Comment:
Wouldn't it be better to check as soon as possible type consistency for the column type and the value type, instead of waiting for appending a value of the wrong type?